homeassistant/components/vacuum/roomba.py

```diff
@@ -90,7 +92,14 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     )
     _LOGGER.info("Initializing communication with host %s (username: %s)",
                  host, username)
-    yield from hass.async_add_job(roomba.connect)
+
+    try:
+        with async_timeout.timeout(10):
```

Comment:
Make timeout 9 seconds and instead of returning False, raise PlatformNotReady. Home Assistant will then try again in the future. Reason it's 9 seconds btw is so that you avoid the slow platform warning.